### PR TITLE
feat(36266): async function needs Promise<T> return type quick fix

### LINKED ISSUE #36266: async function needs Promise<T> return type quick fix
## Search Terms

- return type quick fix
- async function quick fix
- global promise type

## Suggestion

It would be nice if TypeScript contributed a quick fix to resolve `The return type of an async or global function must be the global Promise<T> type` errors. The fix would simply wrap the existing return type into a Promise.

## Checklist

My suggestion meets these guidelines:

* [x ] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [ x] This wouldn't change the runtime behavior of existing JavaScript code
* [ x] This could be implemented without emitting different JS based on the types of the expressions
* [x ] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x ] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #36266

### CODE REVIEW

**Comment by ExE-Boss** [comment]
`PromiseLike<foo>` should get auto‑fixed to `Promise<foo>` instead of the current `Promise<PromiseLike<foo>>`, which is incorrect: <https://github.com/microsoft/TypeScript/issues/27711>.

**Comment by ExE-Boss** [comment]
This should probably also have some tests for non‑`void` custom `PromiseLike`s.

File: src/services/codefixes/fixReturnTypeInAsyncFunction.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        getCodeActions: context => {
+            const { sourceFile, program, span } = context;
+            const checker = program.getTypeChecker();
+            const info = getInfo(sourceFile, program.getTypeChecker(), span.start);
+            if (!info) {
+                return undefined;
+            }
+            const { returnTypeNode, returnType, promisedTypeNode, promisedType } = info;
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, sourceFile, returnTypeNode, promisedTypeNode));
+            return [createCodeFixAction(fixId, changes, [Diagnostics.Replace_0_with_Promise_1, checker.typeToString(returnType), checker.typeToString(promisedType)], fixId, Diagnostics.Fix_all_incorrect_return_type_of_an_async_functions)];
```
Comment (elibarzilay) [inline src/services/codefixes/fixReturnTypeInAsyncFunction.ts:27]: Picky: would you mind breaking up this huge line?

File: tests/baselines/reference/api/typescript.d.ts [hunk 2]
```diff
@@ -2035,6 +2035,7 @@ declare namespace ts {
         getBaseTypes(type: InterfaceType): BaseType[];
         getBaseTypeOfLiteralType(type: Type): Type;
         getWidenedType(type: Type): Type;
+        getAwaitedType(type: Type): Type | undefined;
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/api/typescript.d.ts:2038]: I think we'd want this to be internal.

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-3.9

**Pull request merged**